### ISSUE TITLE
Improve the aethetics of the emails we are sending

### ISSUE THREAD

**Issue by danhgov** [bcgov/MFIN-Data-Catalogue#516]
Currently, all of our emails are based on this very simple template that has been uploaded to GC Notify:
    ![image](https://github.com/bcgov/MFIN-Data-Catalogue/assets/130506582/4071f1b9-a112-4bc2-8ab6-5b5df7608f89)

This ends up with emails that look like this:
    ![image](https://github.com/bcgov/MFIN-Data-Catalogue/assets/130506582/da79e2c4-fd45-420e-bf92-cc1559f275f4)

I don't think we need the province's logo to be there at all (I'm guessing it was added by OpenPlus), as most people receiving these are part of the BC Gov, or at least have an IDIR. I'd suggest that it:

- be a much smaller logo, if at all
- be for the Ministry of Finance, not the Government of BC
- be in the footer, not the top of the email
- include some text about the Data Catalog:
  - a link to the main DC page
  - the text "Ministry of Finance Data Catalogue"
  - perhaps the name of a person to contact
  - etc

In addition to the footers of these emails, I suggest we might want to rework the content. Currently it is very abrupt: "Review needed:", and a list of links to click. Might be nice to have "To uphold the trust of our users in the Data Catalogue, it is important that the records be reviewed periodically..." etc. 

In addition, rather than just listing which records need review, it might be nice to also include the last time each was reviewed, and how often they are set to review. e.g. "Database XYZ - should be reviewed every 6 months, was last reviewed Jan 15, 2024. Next review is due 18 days from now, on July 15."

And these are just the ideas of a lowly developer. It might be nice to get Srushti or Vanessa to weigh in on this, too. :-) 

([Sister ticket, in Teams](https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/mytasks?tenantId=6fdb5200-3d0d-4a8a-b036-d3685e359adc&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fv1%2Fplan%2FhwqO5l_JEk6qxja33KktamUACvvJ%2Ftask%2Fk4P66H6FL0ydxW7kjtzQSGUAGHMh%22%7D))

**Comment by NicoledeGreef** [comment]
I like your ideas, @danhgov and I will flag this one for a UX consult as you mentioned.

**Comment by danhgov** [comment]
There is also the other email:

Subject: An asset you have bookmarked has been updated
Body: The following asset has been updated: \<Link to data-set XYZ>

Issues to resolve:

* There is needless redundancy between subject and body
* There is no introduction, reminding the recipient why they might care about the update
* The term "Asset" is needlessly generic and doesn't match the terminology used in the site. I think it should be called a "Metadata Record" or a "Data Set".
 
Here are a couple ideas for how to rewrite.
  * "IMB just made updates to CHEFS Form 'XYZ' data set, which you have bookmarked."
  * The Oracle report 'ABC' was just updated by the Property Taxation Branch in the Finance Data Catalogue. You have bookmarked this metadata record, which is why we are letting you know about it.

**Comment by vmyeong** [comment]
Hi @NicoledeGreef , @danhgov I got this task through Victoria. I will work on the email mockup once I'm back on August 7th based on Dan's comments and additional user-friendly elements I think are necessary. The mockup won't be built in HTML format (which is beyond my capability), but it can be used as a guideline. The mockup will illustrate the general layout, so the wording for each case won't be part of it. I will use this GitHub page as the main source of communication. Let me know if there's anything I should be aware of before jumping into the design.

**Comment by vmyeong** [comment]
@NicoledeGreef @danhgov  The design suggestion for the email notification is below. Let me know what you think.
![Data Catalogue Email Template](https://github.com/user-attachments/assets/82e3fa55-9a03-496b-a172-2ad13e4dd522)

**Comment by NicoledeGreef** [comment]
@vmyeong That is a sweet design - thank you!

I would tweak the first paragraph slightly to read:

"To uphold the trust of consumers, it is important that your metadata records be reviewed periodically. Your attention to this task ensures the reliability of the metadata we host in the Finance Data Catalogue and supports discovery and appropriate use of the Ministry's data holdings."

**Comment by vmyeong** [comment]
@NicoledeGreef @danhgov 
Yes, feel free to adjust the verbiage as needed. Mine is just a placeholder :) Regarding the design and layout, do you see any changes that should be made? If everything looks good, I'll mark this task as completed. Thanks!

**Comment by danhgov** [comment]
@vmyeong, that looks _amazing_!! Much better than I had envisioned. Thank you!

We need a **second email template**, though (if you agree, @NicoledeGreef). See [my comment above from June 27](https://github.com/bcgov/MFIN-Data-Catalogue/issues/516#issuecomment-2195699181).

And for _both_ of these email templates, we also need to decide on the text of the email's **subject** line.

**Comment by danhgov** [comment]
@NicoledeGreef, do you want me to go ahead and implement this? If so, I will start.

Note that there are _two_ templates needed:

- "Review needed" (the one the Vanessa already mocked up for us), and also
- **"Some MD Records that you have bookmarked have been updated"**

This second template still needs to be designed.

**Comment by vmyeong** [comment]
@danhgov @NicoledeGreef Just to confirm, does "MD" refer to "metadata" in this context? Additionally, could you provide a few more details? Such as..:
The email says "some" records. How many updated records should we display directly in the email (what's the maximum)?
Besides the option for users to click the link and check the updated MD records, are there any other functions or actions we should include in the email?
Once the content is finalized, I can create a mockup for your review.

**Comment by danhgov** [comment]
Sorry, @vmyeong, I wasn't very clear. Yes, "MD" refers to "metadata". But please mostly ignore the text I wrote-- it was just to flag to @NicoledeGreef which other email I was talking about, not attempting to spec anything out for it.

Also, I forgot that those emails are sent out one by one in "real time", whenever a record is updated -- there is _not_ a summary email sent with all the records you've bookmarked that have been updated recently.

Here is what one of those emails typically looks like, at present. The blue hyperlink is the name of the asset.
![image](https://github.com/user-attachments/assets/7e492a8f-1f56-4c8f-a3e3-100d484ecbbf)

Also, I love your idea that people might want other functions. I'd suggest an "unbookmark" link be added to the email, with text something like "I don't need to see updates to this asset anymore".

I'd also suggest that if the asset is a Metadata Record, that we use that word: "The following Metadata Record has been updated". As I said earlier in this ticket, I think that using the word "asset" is needlessly generic in this context.

@NicoledeGreef, are there other types of assets that people can also bookmark? If so, are there any aspects of them that might need a different treatment in these emails than the Metadata Records do?

**Comment by NicoledeGreef** [comment]
@NicoledeGreef reminder to ask for the assets for the email as well as a new graphic for Subscriber Alerts.
Need copy for the Subscriber Alerts and also add in Remove Bookmark option for assets listed in Subscriber Alert email to save users a step in case they truly don't want to be notified about changes to items they've Bookmarked.

**Comment by danhgov** [comment]
We talked about this today. Dan will:

- Reach out to Vanessa for anything more that is needed -- such as the graphic assets needed.
- Add a button to unsubscribe.
- Look over other fields from the asset, and possibly add more of them (besides just the title).
- @NicoledeGreef will then look this over, and approve or change it.

**Comment by vmyeong** [comment]
@NicoledeGreef @danhgov This is an MD update notification email mockup I created based on the directions above. More bullet points can be added for each MD if you think it’s necessary. The verbiage is just a draft, so feel free to make changes. 
![Data Catalogue Email Template_MD Updated](https://github.com/user-attachments/assets/cde03b6a-b209-4390-b7ac-576b725502d2)

**Comment by danhgov** [comment]
This is awesome - thanks @vmyeong!

**Comment by vmyeong** [comment]
![update_icon](https://github.com/user-attachments/assets/b6886d4d-2445-4580-b918-03263a38ce21)
![cancel_book mark](https://github.com/user-attachments/assets/aff28ccd-dd3f-4074-940b-4ed2cc525657)
![mfin_imb logo_horizontal](https://github.com/user-attachments/assets/9f16db00-69fc-4bb3-8e8c-dad0b3522a5a)
![review_needed](https://github.com/user-attachments/assets/2df323c2-6728-42d8-995e-3d3908294331)

**Comment by vmyeong** [comment]
@danhgov As you requested, I uploaded some icons. Let me know if you need different format.

**Comment by danhgov** [comment]
@NicoledeGreef  and @vmyeong, I am sorry to report that, upon starting to figure out how to implement Vanessa's mockups, I've learned that the framework provided by GC Notify (Government of Canada Notification) that we are using to send emails is, in fact, very _very_ limited, in terms of us being able to make things look the way we want. 

https://notification.canada.ca/formatting-emails

They allow _basic Markdown_ to render headings, subheadings, bullet lists, numbered lists, and a few other basic things.

They _do_ support hyperlinks.

They _do not_ support graphics of any sort, as far as I can see.

There is the large, logo at the top, which they call our "branding":
![image](https://github.com/user-attachments/assets/5ac7beb0-5382-42ab-963a-1886b38ebc9e)

While we do have the option to replace it, even that is tightly controlled. These are the choices we have at present:
![image](https://github.com/user-attachments/assets/22859029-bea2-4bf8-bdfc-f1b64b9a24da)

At the bottom of the page is a link to "request a new logo". That means submitting a new PNG that is exactly 600x108px, and waiting 2 business days for them to approve it.

The GC Notify dashboard shows that an underwhelming number of emails are being sent out from the Data Catalogue:
- In the past week, it has sent 5 emails.
- Our daily usage today saw 1 email sent, out of our maximum quota of 10,000. :-(

So for now, I'll proceed with reworking our emails, in terms of improved, richer _text_-content. But the nice graphics that Vanessa designed for us, I think will only be possible if we abandon the GC Notify platform, and use some in-house service, or send emails directly from the DataCat website.

:-(

**Comment by NicoledeGreef** [comment]
Please work up a header graphic that incorporates the squarish Ministry of Finance logo and the nice modern IMB logo that meets the constraints applied by GC Notify.

**Comment by danhgov** [comment]
@vmyeong, can you revisit this ticket? I'm sorry that we can't use most of your work on the beautiful two designs you did. It turns out the only graphic that our messaging service supports is a 600x108px banner image, which they call the "branding". It will be the same image on all of the emails that are sent.

On our meeting today with Nicole, she said that we should re-do that to include the BC Gov logo, and the IMB one. @NicoledeGreef, should it also include something for the Data Catalogue?

**Comment by danhgov** [comment]
@NicoledeGreef, I am having trouble generating the link to unbookmark an asset directly from the email. The issue is that Drupal has a mechanism to prevent cross-site request forgeries, and this means that a call to the URL to delete a bookmark has a special token (hash value) attached to it, which is linked to your user's current session. When generating the link for the email, we are not logged in as the user who bookmarked the page, and so the token that is generated is the wrong one, and you'll get 'access denied' when you click that link from the email.

I'm sure there is _some_ way around this, but currently it's taking up too much of my time. I'm going to put it on the back burner (and ideally leave it out), unless you feel that this is a feature we need.

**Comment by vmyeong** [comment]
@danhgov I will send you the banner, exactly 600 x 108, with the MFIN and IMB logos together. I’m experiencing Illustrator connection issues today, but I hope to send the banner to you tomorrow.

**Comment by danhgov** [comment]
Thanks, @vmyeong - there is no rush.

@NicoledeGreef, with the old banner, here is what the update-notification emails currently look like on my localhost:
![Image](https://github.com/user-attachments/assets/d21114af-25e8-44d4-a3e5-c704583775fc)

**Comment by danhgov** [comment]
@NicoledeGreef, the review-reminder email is working (on my localhost), and look like this:
![Image](https://github.com/user-attachments/assets/edd9b0b9-c79e-47b2-a7c4-9af8341d40b1)

**Comment by vmyeong** [comment]
@danhgov @NicoledeGreef Uploading logos you can use for the banner. 
![mfin-imb logo02](https://github.com/user-attachments/assets/3a5f0b86-9b86-4731-9b15-ebdea55d3a43)
![mfin-imb logo](https://github.com/user-attachments/assets/9e3c5b07-fa98-4117-9fa3-1496b7abfc09)

**Comment by danhgov** [comment]
These look awesome. @NicoledeGreef, which is your preference?

**Comment by NicoledeGreef** [comment]
Let's try the stacked presentation.

**Comment by danhgov** [comment]
@NicoledeGreef, the new messaging is in place on the Dev/Test servers, for you to try out.

I have changed the wording slightly on the "your bookmarked item has been updated" email, to look now like this:
![image](https://github.com/user-attachments/assets/a8c145bf-de8d-4bd0-b2b9-7660f105c4b9)

I have just realised that the subject-line of "Your bookmarked metadata record has been updated" could be improved, too. I'll get that now.

But in the meantime, you could test-drive the two types of emails. I'm not sure if your admin account has access to [Administer the BC Data Catalogue Module](https://dev.cat.data.fin.gov.bc.ca/admin/config/data-catalogue) or not... if not, just let me know, and I can click the "Send metadata review reminders" button on that page for you, so that you don't need to wait till the weekend.

One other change that is pending: The updated banner image. I uploaded it to GC Notify today. They say "We're reviewing your logo - We’ll implement or request changes to your logo within 2 business days."

**Comment by danhgov** [comment]
@NicoledeGreef, I updated the email's subject.

- Before: "Your bookmarked metadata record has been updated" 
- After: "Update to "<record title>" metadata record"

This is also deployed to Dev/Test.

**Comment by NicoledeGreef** [comment]
@danhgov  - I don't believe i have permissions as a Catalogue admin user to send the notification emails on the fly.

I notice the "For more information or assistance please contact Nicole.deGreef@gov.bc.ca" - can you confirm this is only for **dev** and **test**?  For prod there should be a link to our JSM support portal, no prompt for the recipients to email me. :-) If we cannot differentiate between the notification sent for different environments then we should use the JSM portal across the board.

**Comment by danhgov** [comment]
@NicoledeGreef, you should have received the new-formatted email - sorry I didn't get back you last week with another click of the "send" button.

Currently, the same contact text (your email address) would deploy to all 3 servers. I'll change it to be you on dev/test, and JSM on prod.

**Comment by danhgov** [comment]
@danhgov to swap the banner image to the slightly wider version. And fix the dev/test/prod logic.

**Comment by danhgov** [comment]
@NicoledeGreef, I have submitted the wider logo to GC Notify, for them to approve. I'll swap it in when I hear back from them.

The dev/test/prod logic issue I think had been solved, but simply hadn't finished rolling out when I (impatiently) tested it.

On dev/test, it should say this:
![image](https://github.com/user-attachments/assets/6b040dfc-2093-4f0e-8ddd-5e89027b88c2)

On prod, it should say this:
![image](https://github.com/user-attachments/assets/59df109b-3b8a-4d23-b008-efc6b33d161f)
(... with the link going to https://fincsp.atlassian.net/servicedesk/customer/portal/12)

I can't test it for _certain_ until it actually rolls out to prod (though I am quite sure it should work), so we'll just keep an eye on this to make sure it's correct when the first batch of reminder emails get sent, when it is on prod.

**Comment by danhgov** [comment]
@NicoledeGreef, the new banner image is in place. Emails now look like the screenshot below. 

I propose that we promote this to prod, to test for certain that the contact info in the footer works correctly.

![image](https://github.com/user-attachments/assets/acf8308d-4443-4994-93ae-a2d3b4e8b029)

**Comment by NicoledeGreef** [comment]
Agree - thanks @danhgov
